project-koku/koku
# Update development setup

## Description

- Update the dev setup docs related to setup since I'm going through it now.
- Change the `Makefile` to use Docker Compose v2.
- Enable buildkit so the multi-arch multi-stage image builds properly
- Remove M1 specific compose file and related documentation since it is now covered in the main README

## Testing

1. Checkout Branch
2. `make docker-down`
3. Unset `DOCKER_BUILDKIT` and `compose_file` if they are set and remove them from your `.env` file
4. Run `make docker-up`

A new `koku-base` image should be built that matches the container host architecture

Verify the architecture matches the container host architecture
```
docker ps -a --format 'table {{ .ID }}\t{{ .Names }}\t{{ .Image }}\t{{ .State }}\t{{ .Label "architecture" }}'
```

`docker compose logs -f koku-server koku-worker` should show the server and worker are operating as expected.

## Notes

- [x] Further changes can be made once we simplify the container image build for ARM (see #3909).
